hawkular-inventory-parent/hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/ResourceFilter.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2014-2017 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.inventory.api;
+
+/**
+ * @author Joel Takvorian
+ */
+public class ResourceFilter {
+    private boolean rootOnly;
+    private String typeId;
+    private String feedId;
+
+    public ResourceFilter(boolean rootOnly, String feedId, String typeId) {
+        this.rootOnly = rootOnly;
+        this.feedId = feedId;
+        this.typeId = typeId;
+    }
+
+    public static ResourceFilter rootOnly() {
+        return new ResourceFilter(true, null, null);
+    }
+
+    public static ResourceFilter ofType(String typeId) {
```

Comment:
I don't have any strong opinion, but as @jmazzitelli started to introduce builders, why don't maintain same semantic ?
I see this one has more "lambda" style, and I'm fine just wondering if we want to maintain styles or this is ok.
LGTM in any case.